test/mongodb.test.js

```diff
@@ -823,12 +823,13 @@ describe('mongodb connector', function() {
   });
 
   it('does not execute a nested `$where`', function(done) {
+    const nestedWhereFilter = {where: {content: {$where: 'function() {return this.content.contains("content")}'}}};
     Post.create({title: 'Post1', content: 'Post1 content'}, (err, p1) => {
       Post.create({title: 'Post2', content: 'Post2 content'}, (err2, p2) => {
         Post.create({title: 'Post3', content: 'Post3 data'}, (err3, p3) => {
-          Post.find({where: {content: {$where: 'function() {return this.content.contains("content")}'}}}, (err, p) => {
-            should.not.exist(err);
-            p.length.should.be.equal(0);
+          Post.find(nestedWhereFilter, (err, p) => {
+            should.exist(err);
+            err.message.should.match(/not allowed in query/);
```

Comment:
This is changing what's being asserted in the test. Before, we assert the connector does not report error and returns an empty array. Now you are asserting that an error is reported.

Could you please elaborate more on why are you making this change? I don't know what should be the expected behavior and thus cannot asses impact of this change.

Reply:
I believe the behaviour comes from https://github.com/strongloop/loopback-datasource-juggler/commit/bae60d1c08d6e006f2de4a49df24a2d011d1cea2 because juggler restricts us from performing this type of queries. I also used https://github.com/strongloop/loopback-connector-mongodb/pull/483/files#diff-f44d7a6206dcaa5ddd05b31118b5c2be as inspiration.

Reply:
I'll trust you on this.

Just make sure that all relevant CI builds are passing.

Reply:
a nitpick: I think the test here was intended to make sure when extended operators are allowed, the nested `$where` won't be executed.
 
So could you add the option `{allowExtendedOperators: true}` in the query and assert the error message accordingly?
Just like https://github.com/strongloop/loopback-connector-mongodb/pull/483/files#diff-f44d7a6206dcaa5ddd05b31118b5c2beR786
Otherwise LGTM, thanks!

Reply:
Thanks Janny, added the option and renamed the test accordingly.